x/migration/keeper/msg_server_recover_morse_account.go

```diff
@@ -2,17 +2,123 @@ package keeper
 
 import (
 	"context"
+	"slices"
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
 
+	"github.com/pokt-network/poktroll/x/migration/recovery"
 	"github.com/pokt-network/poktroll/x/migration/types"
+	migrationtypes "github.com/pokt-network/poktroll/x/migration/types"
+	sharedtypes "github.com/pokt-network/poktroll/x/shared/types"
 )
 
 func (k msgServer) RecoverMorseAccount(ctx context.Context, msg *types.MsgRecoverMorseAccount) (*types.MsgRecoverMorseAccountResponse, error) {
 	sdkCtx := sdk.UnwrapSDKContext(ctx)
 
-	// TODO_MAINNET_MIGRATION(@bryanchriswhite): Implement MsgRecoverMorseAccount handler...
-	_ = sdkCtx
+	// Validate morese account recovery message.
+	if err := msg.ValidateBasic(); err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
 
-	return &types.MsgRecoverMorseAccountResponse{}, nil
+	// Check if the authority is valid.
+	if k.GetAuthority() != msg.Authority {
+		return nil, status.Error(
+			codes.PermissionDenied,
+			migrationtypes.ErrMorseRecoverableAccountClaim.Wrapf(
+				"invalid authority; expected %s, got %s",
+				k.GetAuthority(), msg.GetAuthority(),
+			).Error(),
+		)
+	}
+
+	// Check if the morse account is listed in the recoverable accounts list.
+	if !slices.Contains(recovery.GetRecoveryAllowList(), msg.GetMorseSrcAddress()) {
```

Comment:
Can you create a helper in the same place where `GetRecoveryAllowList` is defined that just takes in a string and call it something `IsMorseAddressRecoverable`?

Reply:
`IsMorseAddressRecoverable` being enough. I removed the public `GetRecoveryAllowList`